pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -1168,6 +1168,99 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
+						RuleWithOperations: admissionregistrationv1.RuleWithOperations{
+							Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Update},
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{"machine.openshift.io"},
+								APIVersions: []string{"*"},
+								Resources:   []string{"machines"},
+							},
+						},
+					})
+
+				})).Should(Succeed())
+
+				Eventually(k.Object(machineVap), timeout).Should(
+					HaveField("Status.ObservedGeneration", BeNumerically(">=", 2)),
+				)
+
+				By("Updating the VAP binding")
+				policyBinding = &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+				Eventually(k8sClient.Get(ctx, client.ObjectKey{
+					Name: "openshift-only-create-mapi-machine-if-authoritative-api-capi"}, policyBinding), timeout).Should(Succeed())
+
+				Eventually(k.Update(policyBinding, func() {
+					// We want to have our paramref reference the CAPI namespace,
+					// since we `GenerateName` it is not static
+					policyBinding.Spec.ParamRef.Namespace = capiNamespace.GetName()
+					// We need to update the namespace in our namespaceSelector,
+					// since also use `GenerateName` here
+					policyBinding.Spec.MatchResources.NamespaceSelector.MatchLabels = map[string]string{
+						"kubernetes.io/metadata.name": mapiNamespace.GetName(),
+					}
+				}), timeout).Should(Succeed())
+
+				// Wait until the binding shows the patched values
+				Eventually(k.Object(policyBinding), timeout).Should(
+					SatisfyAll(
+						HaveField("Spec.MatchResources.NamespaceSelector.MatchLabels",
+							HaveKeyWithValue("kubernetes.io/metadata.name",
+								mapiNamespace.GetName())),
+					),
+				)
+
+				checkVAPCapiMachine := clusterv1resourcebuilder.Machine().WithGenerateName("capi-vap-checking-machine").WithNamespace(capiNamespace.Name).Build()
+				Expect(k8sClient.Create(ctx, checkVAPCapiMachine)).To(Succeed())
+
+				Eventually(k.Get(checkVAPCapiMachine)).Should(Succeed())
+
+				By("Creating a throwaway MAPI machine")
+				testMachine := mapiMachineBuilder.WithGenerateName("test-machine").Build()
+				Eventually(k8sClient.Create(ctx, testMachine), timeout).Should(Succeed())
+
+				By("Setting the throwaway MAPI machine AuthoritativeAPI to Cluster API")
+				Eventually(k.UpdateStatus(testMachine, func() {
+					testMachine.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityClusterAPI
+				})).Should(Succeed())
+
+				Eventually(k.Object(testMachine), timeout).Should(
+					HaveField("Status.AuthoritativeAPI", Equal(machinev1beta1.MachineAuthorityClusterAPI)))
+
+				Eventually(k.Update(testMachine, func() {
+					testMachine.ObjectMeta.Labels = map[string]string{"test-sentinel": "fubar"}
+				}), timeout).Should(MatchError(ContainSubstring("policy in place")))
+			})
+
+			It("Doesn't allow creation of the MAPI machine with the same name", func() {
+				Eventually(func() error {
```

Comment:
We don't need this `Eventually( func() error)` approach that recreates a fresh capi/mapi machine pair with every retry, as we have the sentinel proving that the VAP is enforced. 

We should hit this on the first create, once the before each passes, so this can just become a single attempt to create :)